packages/flutter_tools/test/src/common.dart

```diff
@@ -137,3 +139,59 @@ Future<void> expectToolExitLater(Future<dynamic> future, Matcher messageMatcher)
     fail('ToolExit expected, got $e\n$trace');
   }
 }
+
+@isTest
+void testWithoutContext(String description, FutureOr<void> body(), {
```

Comment:
This needs some comments about when to use and/or not use this. I think it would also be good to have a code health issue filed on github linked from here that describes what the next steps are, and maybe suggests some candidate components that could benefit.

Reply:
Filled https://github.com/flutter/flutter/issues/47161 and added a doc comment